### PR TITLE
Ribbon minor fixes

### PR DESCRIPTION
**Pull request by mdantonio**
Some minor things that I propose to change:

- Rename type arguments into data_type
(type is a built-in function, should not be used as a parameter)

- Change how boundsLower is calculated
(equivalent expression but I find it more clear)

- Remove .keys() for dict iterations and assertions
(in operator already search between keys and for operator loop over keys
by default, so no need to create a .keys iterator

- Replace list(dict.keys()) + sort with sorted(dict.keys())
(equivalent but shorter and faster)

- Remove a double for loop (the 1st used to initialize some dictionaries)

- Remove a print(length) that was probably some old debug code

- Replace string concatenation in a loop with the join operator

P.S. I would like to ask a review to Ken, but I can't find his username on github

### CODE REVIEW

File: ribbon/ribbon/encode.py [hunk 1]
```diff
@@ -37,47 +37,47 @@
...
     signed = match[1] == ''
     width = int(match[2]) if len(match) == 3 else 256
 
     if width % 8 != 0 or width > 256 or (len(match) == 3 and match[2] != str(width)):
-        raise ValueError(f'Invalid numeric width: {type}')
+        raise ValueError(f'Invalid numeric width: {data_type}')
 
     boundsUpper = 2 ** (width - 1) - 1 if signed else 2 ** (width) - 1
-    boundsLower = (boundsUpper + 1) * (-1)
+    boundsLower = -boundsUpper - 1
```
Comment (pdonorio) [inline ribbon/ribbon/encode.py:58]: personally I understand better the older version :D

**Review by pdonorio — APPROVED** [review]

**Comment by pdonorio** [comment]
> P.S. I would like to ask a review to Ken, but I can't find his username on github

@mdantonio you may share the PR link in slack to them

File: ribbon/ribbon/encode.py (hunk 1 above)
Comment (mdantonio) [reply ribbon/ribbon/encode.py:58]: Really? So the math is particularly subjective! 🙂
In this case let Ken to decide if maintain the current version or accept the change, definitely it is not an important change and I can quickly revert it 😉

**Review by kenchangh — APPROVED** [review]
Looks good to me

File: ribbon/ribbon/encode.py (hunk 1 above)
Comment (kenchangh) [reply ribbon/ribbon/encode.py:58]: This seems fine

Comment (pdonorio) [reply ribbon/ribbon/encode.py:58]: cool

**Pull request merged**